spateo/digitization/contour.py

```diff
@@ -0,0 +1,276 @@
+"""Written by @Jinerhal, adapted by @Xiaojieqiu.
+"""
+
+import random
+from typing import List, Optional, Tuple, Union
+
+import cv2
+import numpy as np
+from anndata import AnnData
+from nptyping import NDArray
+from skimage import morphology
+
+from ..configuration import SKM
+from ..logging import logger_manager as lm
+
+
+@SKM.check_adata_is_type(SKM.ADATA_UMI_TYPE, optional=True)
+def gen_cluster_image(
+    adata: AnnData,
+    bin_size: int = None,
+    spatial_key: str = "spatial",
+    cluster_key: str = "scc",
+    label_mapping_key: str = "cluster_img_label",
+    show: bool = True,
+    cmap: str = "tab20",
+):  # need output type
```

Comment:
Please help specify var-types for function parameters and returns.

Reply:
defintively